### PR TITLE
Add data plane CloudFormation stack

### PR DESCRIPTION
**Pull request by hectcastro**
The data plane consists of an PostgreSQL RDS database and a Redis ElastiCache cache cluster. Each data store is accompanied by CloudWatch alarms, but only the PostgreSQL RDS database has been setup with a Route 53 private DNS record. The private DNS record for Redis will come with #339.

In addition, this pull request moves the bastion/monitoring server creation into the data plane stack. The security group for the bastion/monitoring server has been updated, and public/private Route 53 DNS records get created for it.

Connects #295
Connects #338

Also, tagging @lliss and @kdeloach.

---

This pull request takes a while to test because spinning up the RDS database and cache cluster is a long process. After it completes successfully, there should be a new PostgreSQL RDS database, Redis ElastiCache cache cluster, and a handful of new Route 53 DNS records.

### CODE REVIEW

File: deployment/cfn/utils/constants.py [hunk 1]
```diff
@@ -15,6 +15,13 @@
...
 
-SSH = 22
+GRAPHITE = 2003
+GRAPHITE_WEB = 8080
 HTTP = 80
 HTTPS = 443
+KIBANA = 5601
+POSTGRESQL = 5432
+REDIS = 6379
+RELP = 20514
```
Comment (lliss) [inline deployment/cfn/utils/constants.py]: What is RELP?

Comment (hectcastro) [inline deployment/cfn/utils/constants.py]: http://en.wikipedia.org/wiki/Reliable_Event_Logging_Protocol

File: deployment/cfn/data_plane.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+            VpcId=Ref(self.vpc_id),
+            SecurityGroupIngress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [POSTGRESQL]
+            ],
+            SecurityGroupEgress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
```
Comment (lliss) [inline deployment/cfn/data_plane.py]: Do we need the postgres port open for outbound traffic?

Comment (hectcastro) [inline deployment/cfn/data_plane.py]: Nope, good catch. Addressed in https://github.com/WikiWatershed/model-my-watershed/commit/ff9e2268465a71a4602eb154d6a6c38cbe676d31.

**Comment by lliss** [comment]
Aside for the issue mentioned above. +1. I've run this and it does spin up the data storage stack as expected.

**Pull request merged**